app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -135,18 +125,10 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         val context = inflater.context
         val layout = inflater.inflate(R.layout.fragment_browser, container, false)
 
-        val viewModelFactory = context.serviceLocator.viewModelFactory
-        CursorController.newInstanceOnCreateView(
-            this,
-            cursorParent = layout.browserFragmentRoot,
-            view = layout.cursorView,
-            viewModel = ViewModelProviders.of(this, viewModelFactory).get(CursorViewModel::class.java)
-        ).also { (controller, disposable) ->
-            this.cursor = controller
-            onCreateViewCompositeDisposable.add(disposable)
-            lifecycle.addObserver(controller)
-            context.serviceLocator.cursorEventRepo.setCursorController(controller)
+        layout.browserFragmentRoot.addOnLayoutChangeListener { _, _, _, right, bottom, _, _, _, _ ->
+            context.serviceLocator.cursorController.screenBounds = PointF(right.toFloat(), bottom.toFloat())
         }
+        context.serviceLocator.cursorEventRepo.setWebViewCouldScrollInDirection { layout.engineView.couldScrollInDirection(it) }
```

Comment:
nit: -> `canWebViewScrollInDirectionProvider` or similar: I find it confusing to distinguish between references that are values and functions that provide those values since they often get named the same thing.

Also (with old naming convention for time), probably not that important but:
```kotlin
context.serviceLocator.cursorEventRepo.setWebViewCouldScrollInDirection(layout.engineView::couldScrollInDirection)
```

Or even:
```kotlin
context.serviceLocator.cursorEventRepo.webViewCouldScrollInDirection = layout.engineView::couldScrollInDirection
```